add invalid property error

lshig opened the pull request:
Closes https://github.com/mapbox/batfish/issues/45.

Properties are documented here: https://github.com/mapbox/batfish/blob/master/docs/configuration.md.

Cut a ticket for documentation on `verbose` https://github.com/mapbox/batfish/issues/98

Used existing `lodash` library to compare 2 properties name lists and return the list of properties that don't match.

davidtheclark commented on lib/validate-config.js:
```diff
@@ -56,6 +56,51 @@ function validateConfig(config, configPath) {
+    'production',
+    'port',
+    'verbose'
+  ];
```
Let's pull this list out of the function, since it doesn't have to be created when the function executes — but is a kind of constant. Also, let's add a comment noting that whenever you add a new configuration property you'll need to add to this list.

davidtheclark commented on lib/validate-config.js:
```diff
@@ -56,6 +56,51 @@ function validateConfig(config, configPath) {
+  const invalidProperties = _.pullAll(
+    configWithDefaultsProperties,
+    validConfigProperties
+  );
```
Nice method for getting the full list of invalid properties, instead of just error on the first one!

lshig commented:
I moved the constants in 4b8cb15 and added tests in 1ac1f7d. Let me know if I missed anything or if you have anymore questions/feedback @davidtheclark 😸
